src/service_account.rs

```diff
@@ -115,10 +115,11 @@ impl JWT {
     fn sign(&self, private_key: &str) -> Result<String, Box<error::Error>> {
         let mut jwt_head = self.encode_claims();
         let key = try!(decode_rsa_key(private_key));
-        let signer = try!(sign::RSASigner::new(&key)
-            .map_err(|_| io::Error::new(io::ErrorKind::Other, "Couldn't initialize signer")));
-        let signature = try!(signer.sign(rustls::SignatureScheme::RSA_PKCS1_SHA256,
-                                         jwt_head.as_bytes())
+        let signing_key = try!(sign::RSASigningKey::new(&key)
+            .map_err(|_| io::Error::new(io::ErrorKind::Other, "Couldn't initialize signing key")));
+        let signer = try!(signing_key.choose_scheme(&[rustls::SignatureScheme::RSA_PKCS1_SHA256])
```

Comment:
Something tells me that declaring such a reference in place may be an anti-pattern, but Rust didn't yell at me, so I decided to leave it like this for now.

Reply:
It's not unusual. You need a slice here and it just so happens that your slice is static.